Assets/Scripts/Algorithms/Patrolling/CognitiveCoordinated/CognitiveCoordinatedBase.cs

```diff
@@ -11,15 +11,21 @@
 
 namespace Maes.Algorithms.Patrolling
 {
-    public sealed class CognitiveCoordinated : PatrollingAlgorithm
+    public abstract class CognitiveCoordinatedBase : PatrollingAlgorithm
     {
-        public override string AlgorithmName => "Cognitive Coordinated Algorithm";
+        protected const int VirtualStigmergyMessageId = 0;
         private List<Vertex> _currentPath = new();
         private int _pathStep;
         private readonly Dictionary<(int, int), Vertex[]> _pathsCache = new();
 
         protected override bool AllowForeignVertices => true;
 
+        /// <summary>
+        /// The "global" map that is used to coordinate the robots.
+        /// Whether the map is global or not depends on the implementation of the algorithm.
+        /// </summary>
+        protected virtual PatrollingMap _globalMap => null!;
```

Comment:
We should have _globalMap in the static Coordinator version.

Reply:
We must not use global map for the virtual stigmergy version.

Reply:
Yep, it's removed now